### PR TITLE
Chore: Skip import from hosts folder

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changelog Description
Do not look for addons in hosts folder.

## Additional info
Removed remaining `__init__.py` from empty hosts dir.

### CODE REVIEW

File: client/ayon_core/addon/base.py [hunk 1]
```diff
@@ -411,82 +411,56 @@ def _load_addons_in_core(
...
-                        )
-                        sys.modules[new_import_str] = default_module
-                        setattr(openpype_modules, basename, default_module)
-
-                    except Exception:
-                        log.warning(
-                            "Failed to import host folder {}".format(basename),
-                            exc_info=True
-                        )
+        elif ext not in (".py", ):
```
Comment (BigRoy) [inline client/ayon_core/addon/base.py:445]: Why not just this?
```suggestion
        elif ext != ".py":
```

File: client/ayon_core/addon/base.py [hunk 2]
```diff
@@ -411,82 +411,56 @@ def _load_addons_in_core(
...
-        is_in_modules_dir = dirpath == modules_dir
-        ignored_filenames = set()
-        if is_in_modules_dir:
-            ignored_filenames = set(IGNORED_DEFAULT_FILENAMES)
-
-        for filename in os.listdir(dirpath):
-            # Ignore filenames
-            if filename in IGNORED_FILENAMES or filename in ignored_filenames:
-                continue
+    ignored_filenames = set(IGNORED_DEFAULT_FILENAMES)
```
Comment (BigRoy) [inline client/ayon_core/addon/base.py:421]: This variable could just be a `set` on the module level constant already looking at the usages?

Same for `IGNORED_FILENAMES` maybe?

File: client/ayon_core/addon/base.py [hunk 3]
```diff
@@ -411,82 +411,56 @@ def _load_addons_in_core(
...
 
-            fullpath = os.path.join(dirpath, filename)
-            basename, ext = os.path.splitext(filename)
+    for filename in os.listdir(modules_dir):
+        # Ignore filenames
+        if filename in IGNORED_FILENAMES or filename in ignored_filenames:
```
Comment (BigRoy) [inline client/ayon_core/addon/base.py:425]: If these are both sets might be worth merging it before the for loop to make this just:

```python
ignored_filenames = IGNORED_DEFAULT_FILENAMES + IGNORED_FILENAMES
if filename in ignored_filenames:
    continue
```

But maybe those two constants (now being constants?) could be merged on the module level already as well into one set.

File: client/ayon_core/addon/base.py [hunk 4]
```diff
@@ -411,82 +411,56 @@ def _load_addons_in_core(
...
+        # TODO add more logic how to define if folder is addon or not
+        # - check manifest and content of manifest
+        try:
+            # Don't import dynamically current directory modules
+            new_import_str = "{}.{}".format(modules_key, basename)
+
+            import_str = "ayon_core.modules.{}".format(basename)
+            default_module = __import__(import_str, fromlist=("", ))
+            sys.modules[new_import_str] = default_module
+            setattr(openpype_modules, basename, default_module)
```
Comment (BigRoy) [inline client/ayon_core/addon/base.py:457]: Weren't we removing `openpype_modules` elsewhere already? :) I thought it was already gone.

Comment (iLLiCiTiT) [reply client/ayon_core/addon/base.py:457]: Not safe to remove yet.

File: client/ayon_core/addon/base.py (hunk 3 above)
Comment (BigRoy) [reply client/ayon_core/addon/base.py:425]: Cool - fine with me.

File: client/ayon_core/addon/base.py (hunk 4 above)
Comment (iLLiCiTiT) [reply client/ayon_core/addon/base.py:457]: I just realized this happened only for modules living inside `modules` dir, so we can remove it.

**Review by kalisp — APPROVED** [review]
Seems to work, tried to sneak in code from addon back to hosts, didnt get picked up.

**Pull request merged**